ACE/ace/OS_NS_sys_utsname.cpp

```diff
@@ -214,13 +214,17 @@ ACE_OS::uname (ACE_utsname *name)
 
   return ACE_OS::hostname (name->nodename, maxnamelen);
 #elif defined (ACE_INTEGRITY)
+# if defined (ACE_LACKS_GETHOSTNAME)
+  ACE_UNUSED_ARG (name);
```

Comment:
Why make this exception here, doesn't ACE_OS::hostname handle this?

Reply:
Calling ACE_OS::hostname causes an infinite calling loop between ACE_OS::hostname and ACE_OS::uname and gethostname is never called.